tests/e2e/client/apiproxy_bookstore_quota_test.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import utils
+import sys
+import time
+from utils import ApiProxyClientTest
+
+class C:
+  pass
+FLAGS = C
+
+class ApiProxyBookstoreTest(ApiProxyClientTest):
+  """End to end integration test of bookstore application with deployed
+  ESP at VM.  It will call bookstore API according its Swagger spec
+  1) set quota limit to 30
+  2) send traffic 60 qpm for 150 seconds and count response code 200
+  3) check count between 45 to 135
+  """
+
+  def __init__(self):
+    ApiProxyClientTest.__init__(self, FLAGS.host,
+                           FLAGS.allow_unverified_cert,
+                           FLAGS.verbose)
+
+  def verify_quota_control(self):
+    # turn off verbose log
+    print("Turn off the verbose log flag...");
+    verbose = FLAGS.verbose
+    FLAGS.verbose = False
+    self.set_verbose(FLAGS.verbose)
+
+    # exhaust the quota in the current window.
+    print("Exhaust current quota...");
+    response = self._call_http(path='/quota_read',
+                               api_key=FLAGS.api_key)
+    if response.status_code != 429:
+      while True:
+        response = self._call_http(path='/quota_read',
```

Comment:
Nit: Maybe add a max limit to the number of requests so we don't infinite loop if ESPv2 doesn't track quota correctly? Same for other while True loops here.

Reply:
Done.